index.bs

```diff
@@ -2327,31 +2332,32 @@ To <dfn>obtain a set of possible trigger states</dfn> given a [=randomized respo
         with repetition.
 1. Return |possibleValues|.
 
-To <dfn>obtain a randomized source response pick rate</dfn> given a [=randomized response output configuration=] |config| and a double |epsilon|:
-
-1. Let |possibleValues| be the result of [=obtaining a set of possible trigger states=] with |config|.
-1. Let |numPossibleValues| be the [=set/size=] of |possibleValues|.
-1. Return |numPossibleValues| / (|numPossibleValues| - 1 + e<sup>|epsilon|</sup>).
+To <dfn>obtain a randomized source response pick rate</dfn> given a positive integer |states| and a double |epsilon|:
+1. Return |states| / (|states| - 1 + e<sup>|epsilon|</sup>).
 
 To <dfn>obtain a randomized source response</dfn> given a [=randomized response output configuration=] |config| and a double |epsilon|:
 
 1. Let |possibleValues| be the result of [=obtaining a set of possible trigger states=] with |config|.
-1. Let |pickRate| be the result of [=obtaining a randomized source response pick rate=] with |config| and |epsilon|.
+1. Let |pickRate| be the result of [=obtaining a randomized source response pick rate=] with the [=set/size=] of |possibleValues| and |epsilon|.
 1. Return the result of [=obtaining a randomized response=] with null, |possibleValues|, and
     |pickRate|.
 
 <h3 algorithm id="computing-channel-capacity">Computing channel capacity</h3>
 
-To <dfn>compute the channel capacity of a source</dfn> given a [=randomized response output configuration=] |config| and a double |epsilon|:
-1. Let |pickRate| be the [=obtain a randomized source response pick rate|randomized response pick rate=] with |config| and |epsilon|.
-1. Let |states| be the [=obtain a set of possible trigger states|number of possible trigger states=] with |config|.
+To <dfn>compute the channel capacity of a source</dfn> given a positive integer |states| and a double |epsilon|:
 1. If |states| is 1, return 0.
 1. If |states| is greater than the user agent's [=max trigger-state cardinality=], return an error.
+1. Let |pickRate| be the [=obtain a randomized source response pick rate|randomized response pick rate=] with |states| and |epsilon|.
 1. Let |p| be |pickRate| * (|states| - 1) / |states|.
 1. Return log2(|states|) - h(|p|) - |p| * log2(|states| - 1) where h is the binary entropy function [[BIN-ENT]].
 
 Note: This algorithm computes the channel capacity [[CHAN]] of a q-ary symmetric channel [[Q-SC]].
 
+To <dfn>compute the information gain for attribution scope</dfn> given an integer |numTriggerStates|, a positive integer |attributionScopeLimit|, and a positive integer |maxEventStates|:
+1. Let |totalStates| be |numTriggerStates| + |maxEventStates| * (|attributionScopeLimit| - 1).
+1. If |totalStates| is equal to 0, return 0.
```

Comment:
In what case would `|totalStates|` be 0? Or in other words, would `|numTriggerStates|` ever be 0?

Reply:
|numTriggerStates| should be a positive integer. we made the same assumption that |states| should be positive in the current channel capacity computation : https://wicg.github.io/attribution-reporting-api/#compute-the-channel-capacity-of-a-source

Reply:
If that's the case, why do we need this check? And shall we just say `|numTriggerStates|` is a positive integer?